File: C#/RefactoringWorkshop/RefactoringWorkshop/Bloaters/Primitive/ProductService.cs
using System;
using System.Collections.Generic;

namespace RefactoringWorkshop.Bloaters.Primitive
{
    public class ProductService
    {
        private readonly List<Product> _cart = new List<Product>();

        public bool AddProductToCart(Product product) {
            var productId = product.ProductId;

            if(isValid(productId)) {
                _cart.Add(product);
                return true;
            }
            return false;
        }

        private bool isValid(string productId) {
            return productId.StartsWith("PD");
        }
    }
}

File: Java/RefactoringWorkshop/src/main/java/bloaters/primitive/ProductService.java
package bloaters.primitive;

import java.util.ArrayList;
import java.util.List;

public class ProductService {

    private List<Product> cart = new ArrayList<>();

    public boolean addProductToCart(Product product) {
        String productId = product.getId();

        if(isValid(productId)) {
            return cart.add(product);
        }
        return false;
    }

    private boolean isValid(String productId) {
        return productId.startsWith("PD");
    }
}
